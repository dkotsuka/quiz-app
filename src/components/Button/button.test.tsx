import React from 'react'
import { Button } from '.'
import {create, act} from 'react-test-renderer';


describe("Button", () => {
    const primaryMockFn = jest.fn()
    const secondaryMockFn = jest.fn()
    const primaryBtn = create(<Button.Primary text="Primary" onPress={primaryMockFn}/>)
    const primaryBtnWithIcon = create(<Button.Primary text="Primary" onPress={primaryMockFn} icon="arrowRight"/>)
    const secondaryBtn = create(<Button.Secondary text="Secondary" onPress={secondaryMockFn}/>)

    it("Primary match snapshot", () => {
        expect(primaryBtn).toMatchSnapshot()
    })

    it("Primary with icon match snapshot", () => {
        expect(primaryBtnWithIcon).toMatchSnapshot()
    })

    it("Secondary match snapshot", () => {
        expect(secondaryBtn).toMatchSnapshot()
    })

    it("Primary function is called", () => {
        const touchable = primaryBtn.root.findByProps({testID:"primary-button-touchable"}).props
        act(() => touchable.onPress())
        expect(primaryMockFn).toBeCalled()
    })

    it("Primary function is called", () => {
        const touchable = primaryBtnWithIcon.root.findByProps({testID:"primary-button-touchable"}).props
        act(() => touchable.onPress())
        expect(primaryMockFn).toBeCalledTimes(2)
    })

    it("Secondary function is called", () => {
        const touchable = secondaryBtn.root.findByProps({testID:"secondary-button-touchable"}).props
        act(() => touchable.onPress())
        expect(secondaryMockFn).toBeCalled()
    })
})