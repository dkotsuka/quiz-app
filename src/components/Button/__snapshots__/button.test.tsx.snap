// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Primary match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#4D8AF0",
      "borderRadius": 14,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingVertical": 12,
      "width": "100%",
    }
  }
  testID="primary-button-touchable"
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Roboto Bold",
          "fontSize": 20,
        },
      ]
    }
  >
    Primary
  </Text>
</View>
`;

exports[`Button Primary with icon match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#4D8AF0",
      "borderRadius": 14,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingVertical": 12,
      "width": "100%",
    }
  }
  testID="primary-button-touchable"
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Roboto Bold",
          "fontSize": 20,
        },
      ]
    }
  >
    Primary
  </Text>
  <Image
    source={
      Object {
        "testUri": "../../../assets/icons/arrow-right.png",
      }
    }
    style={
      Object {
        "marginLeft": 12,
      }
    }
  />
</View>
`;

exports[`Button Secondary match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#78809A",
      "borderRadius": 10,
      "borderWidth": 1,
      "height": 44,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
  testID="secondary-button-touchable"
>
  <Text
    style={
      Array [
        Object {
          "color": "#78809A",
          "fontFamily": "Roboto Bold",
          "fontSize": 16,
        },
      ]
    }
  >
    Secondary
  </Text>
</View>
`;
